### PR TITLE
JS-doc : Fix argument and returns types for pages

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Fix argument and returns types for pages
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | no tests needed
| Possible impacts? | none

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25804)

### CODE REVIEW

File: tests/UI/campaigns/data/faker/product.js [hunk 1]
```diff
@@ -62,7 +62,7 @@ class ProductData {
     /** @type {string} Tac rule to apply the product */
     this.taxRule = productToCreate.taxRule || 'FR Taux standard (20%)';
 
-    /** @type {string} Specific price of the product */
+    /** @type {Object|{combinations: ?string, discount: ?number, startingAt: ?number}} Specific price of the product */
```
Comment (NeOMakinG) [inline tests/UI/campaigns/data/faker/product.js:65]: https://jsdoc.app/tags-param.html#optional-parameters-and-default-values

I'm not sure JSDoc got optional param notation

Wdyt about using the TypeScript syntax?
```suggestion
    /** @type {Object|{combinations?: string, discount?: number, startingAt?: number}} Specific price of the product */
```

Do you think that it looks clearer to read?

**Review by NeOMakinG — COMMENTED** [review]
Just a little comment, overall it looks pretty fine to me

File: tests/UI/campaigns/data/faker/product.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/data/faker/product.js:65]: From the official documentation https://jsdoc.app/tags-type.html , I think it's the correct syntax.
image

Comment (NeOMakinG) [reply tests/UI/campaigns/data/faker/product.js:65]: Ok I missed it!

**Review by NeOMakinG — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thank you @boubkerbribri

**Pull request merged**